app/controllers/buyers/applications_controller.rb

```diff
@@ -1,236 +1,27 @@
-
-  def change_plan
-    # there is no need to query available_application_plans as we already have a validation
-    service = @cinstance.service
-    new_plan = accessible_plans.stock.find(plan_id)
-    @cinstance.provider_changes_plan!(new_plan)
-    flash[:notice] = "Plan changed to '#{new_plan.name}'."
-    redirect_to admin_service_application_url(service, @cinstance)
-  end
-
-  def change_user_key
-    with_password_confirmation! do
-      @cinstance.change_user_key!
-      redirect_to admin_service_application_url(@cinstance.service, @cinstance), notice: 'The key was successfully changed'
-    end
-  end
-
-  def destroy
-    if @cinstance.destroy
-      flash[:notice] = 'The application was successfully deleted.'
-      redirect_to admin_buyers_applications_path
-    else
-      flash[:notice] = 'Not possible to delete application'
-      redirect_to :back
-    end
   end
 
   protected
 
-  def applications_context?
-    !params[:account_id]
-  end
-
-  private
-
-  def change_state(*args)
-    action, message = args.shift, args.shift
-    @cinstance.send("#{action}!", *args)
-
-    respond_to do |format|
-      format.html do
-        flash[:notice] = message
-        redirect_to admin_service_application_url(@cinstance.service, @cinstance)
-      end
-
-      format.js do
-        flash.now[:notice] = message
-        render :action => 'update_state'
-      end
-    end
-  end
-
-  def accessible_not_bought_cinstances
-    current_user.accessible_cinstances.not_bought_by(current_account)
-  end
-
-  def find_cinstance
-    @cinstance = accessible_not_bought_cinstances
-                  .includes(plan: [:service, :original, :plan_metrics, :pricing_rules])
-                  .find(params[:id])
-    @account = @cinstance.account
-  end
-
-  def find_buyer
-    account_id = params[:account_id]
-    return unless account_id
-
-    @account = current_account.buyers.find(account_id)
-  end
-
-  def find_plans
-    @plans = accessible_plans.stock
-  end
-
-  def find_service(id = params[:service_id])
-    @service = accessible_not_bought_cinstances.find(id) if id
-  end
-
-  def find_provider
-    @provider = current_account
-  end
-
-  def accessible_services
-    @accessible_services ||= current_user.accessible_services.includes(:application_plans)
-  end
-
-  helper_method :accessible_services
-
-  def accessible_plans
-    current_account.application_plans.where(issuer: accessible_services)
-  end
-
-  def plan_id
-    @plan_id ||= params.require(:cinstance).permit(:plan_id).tap { |plan_params| plan_params.require(:plan_id) }[:plan_id]
-  end
-
-  def authorize_partners
-    authorize! :manage, :partners
-  end
-
   def authorize_multiple_applications
-    authorize! :manage, :multiple_applications if @account.has_bought_cinstance?
-  end
-
-  module AccountForNewPlan
-    # dummy methods for formstastic
-    def service_plan_id; end
-  end
-
-  def extend_cinstance_for_new_plan
-    @cinstance.extend(AccountForNewPlan)
+    find_buyer
```

Comment:
Exactly the same as https://github.com/3scale/porta/pull/2470/files?file-filters%5B%5D=.rb#r649798379